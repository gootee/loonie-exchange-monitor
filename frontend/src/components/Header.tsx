const Header = () => {
  return <header className="row header">
    <div className="text-center d-flex justify-content-center align-items-center">
      <div className="h1Font">Loonie Exchange Monitor</div>
    </div>
  </header>
}

export default Header