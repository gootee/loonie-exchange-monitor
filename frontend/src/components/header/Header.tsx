import React from 'react'
import './header.css'
interface IHeaderProps {
  title: string
}

function Header (props: IHeaderProps) {
  return <header  data-testid="header" className="row header">
    <div className="text-center d-flex justify-content-center align-items-center">
      <div className="h1Font">{ props.title }</div>
    </div>
  </header>
}

export default Header