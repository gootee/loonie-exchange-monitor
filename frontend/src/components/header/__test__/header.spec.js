import React from "react"
import ReactDom from 'react-dom'
import Header from '../Header'
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect" 
import renderer from "react-test-renderer"

afterEach(cleanup)

it(("renders without crashing"), () => {
  const div = document.createElement("div")
  ReactDom.render(<Header title="test"></Header>, div)
})

it ("renders header correctly", () => {
  const {getByTestId} = render(<Header title="Click me"></Header>)
  expect(getByTestId('header')).toHaveTextContent("Click me")
})

it ("matches snapshot", () => {
  const tree = renderer.create(<Header title="Loonie Exchange Monitor"></Header>).toJSON()
  expect(tree).toMatchSnapshot()
})