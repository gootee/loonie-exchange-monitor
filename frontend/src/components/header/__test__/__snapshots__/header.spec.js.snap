// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="row header"
  data-testid="header"
>
  <div
    className="text-center d-flex justify-content-center align-items-center"
  >
    <div
      className="h1Font"
    >
      Loonie Exchange Monitor
    </div>
  </div>
</header>
`;
